build-pyg-support.py

```diff
@@ -0,0 +1,257 @@
+
+    return results[0]
+
+
+def build_pytorch(args):
+    print(f"Building PyTorch using the current feedstock with cuda_version={args.cuda}")
+
+    tarfile = None
+    if args.cuda == "none":
+        run_command(
+            [
+                str((args.feedstock_dir / "build-locally.py").absolute()),
+                str(
+                    find_file(
+                        args.feedstock_dir / ".ci_support", f"linux_64_*cuda_compiler_versionNone*python{args.python}*"
+                    ).name
+                )[:-5],
+            ]
+        )
+
+        tarfile = find_file(
+            args.feedstock_dir / "build_artifacts" / "linux-64",
+            f"pytorch-{PACKAGES_META['torch'].tag}-cpu_py{args.python.replace('.','')}*_openmpi.*.bz2",
+        )
+
+    else:
+        run_command(
+            [
+                str((args.feedstock_dir / "build-locally.py").absolute()),
+                str(
+                    find_file(
+                        args.feedstock_dir / ".ci_support",
+                        f"linux_64_*cuda_compiler_version{args.cuda}*python{args.python}*",
+                    ).name
+                )[:-5],
+            ]
+        )
+
+        tarfile = find_file(
+            args.feedstock_dir / "build_artifacts" / "linux-64",
+            f"pytorch-{PACKAGES_META['torch'].tag}-cuda{args.cuda.replace('.', '')}"
+            + f"py{args.python.replace('.','')}*_openmpi.*.bz2",
+        )
+
+    destination_path = pathlib.Path.home() / "conda-bld" / "linux-64/"
+    shutil.copy(tarfile, destination_path)
+
+    print("PyTorch built")
+    return destination_path / tarfile.name
+
+
+def get_line_after_match(file, match):
+    lines = None
+    with open(file, "r") as read_file:
+        lines = read_file.readlines()
+
+    for line in lines:
+        index = line.find(match)
+        if index != -1:
+            return line[index + len(match) :].strip()
+    raise RuntimeError(f"Match '{match}' not found in file '{file}'.")
+
+
+class GitClone:
+    """
+    An instance of this class represents a remote repo cloned to the specified
+    directory. It is recommended to use tmpfile with this class to both avoid
+    namespace errors, and allow the OS to handle directory removal.
+    """
+
+    def __init__(self, directory, package):
+        self.package = package
+        self.directory = directory
+
+        tag = PACKAGES_META[package].tag
+        if tag == "*":
+            tag = "master"
+        run_command(["git", "clone", "-b", tag, PACKAGES_META[package].url, str(self.directory)])
+        self.conda_dir = self.directory / "conda" / PACKAGES_META[package].name
+
+    def get_version(self):
+        return get_line_after_match(self.conda_dir / "meta.yaml", "version: ")
+
+    def apply_patch(self, args):
+        os.chdir(self.directory)
+        for patch in os.listdir(args.feedstock_dir / "pyg_support_patches" / self.package):
```

Comment:
NIT: I think there is a method on `Path` that does this more directly. No need for old `os` methods.